src/main/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java

```diff
@@ -181,8 +181,8 @@ public void beginJavadocTree(DetailNode rootAst) {
             final String filePath = getFileContents().getFileName();
             String moduleName = getModuleSimpleName();
             final String checkModuleExtension = "Check";
-            if (moduleName.contains(checkModuleExtension)) {
-                moduleName = moduleName.substring(0, moduleName.indexOf(checkModuleExtension));
+            if (moduleName.endsWith(checkModuleExtension)) {
+                moduleName = moduleName.substring(0, moduleName.lastIndexOf(checkModuleExtension));
```

Comment:
I am good with the changes but since we are checking `endsWith`, does it make more sense if we do `lastIndexOf` or use lengths and subtraction to get the end position?

Reply:
Yes, using lengths and subtraction would be better, updated it.